docs/src/components/productBaseUI/BaseUIThemesDemo.tsx

```diff
@@ -141,12 +154,20 @@ const StyledSelectButton = styled('button')({
   '&:hover': {
     backgroundColor: 'var(--Tab-hoverBackground)',
   },
+
+  '&:focus-visible': {
+    outline: '2px solid var(--muidocs-palette-primary-500)',
+    outlineOffset: '2px',
```

Comment:
What do you think about no offset for input like elements?

```suggestion
```

this would get us closer to Joy UI 

<img width="214" alt="Screenshot 2023-08-08 at 16 12 33" src="https://github.com/mui/material-ui/assets/3165635/a74c974e-dfb4-49e4-90e0-fff3105f51ea">

https://mui.com/joy-ui/react-select/

<img width="140" alt="Screenshot 2023-08-08 at 16 12 38" src="https://github.com/mui/material-ui/assets/3165635/546bb594-489a-4304-95d2-22489b871cdd">

https://mui.com/joy-ui/react-button/

Reply:
I've pushed a different focus style, which I think I enjoy more ⎯ let me know what you think!

Reply:
Cool, personally, I think the reason outline 2px on buttons works best while 0px on inputs works best is related to the amount of focus. 2px is much easier to identify. while 0px is harder. So 2px wins for a11y and keyboard UX. 

However, when you are on an input, the focus visible is always shown, and you have your caret. This feels ideal on my end, closer to Joy UI too:

<img width="325" alt="Screenshot 2023-08-14 at 19 53 32" src="https://github.com/mui/material-ui/assets/3165635/706a70c5-bc23-44e7-a91a-2cc139732242">

<img width="350" alt="Screenshot 2023-08-14 at 19 54 05" src="https://github.com/mui/material-ui/assets/3165635/08787ba6-a6ec-4b65-bace-8fa92a56d242">